src/features/QuestionEngine.test.js

```diff
@@ -0,0 +1,295 @@
+import { DateTime } from "luxon";
+import { QuestionEngine } from "./QuestionEngine";
+import { ViewType } from "./ViewType";
+import { ChoiceType } from "./ChoiceType";
+import { StatusType } from "./StatusType";
+import { TitrationType } from "./TitrationType";
+import { QuestionAndAnswer } from "./QuestionAndAnswer";
+import { Question } from "./Question";
+
+describe("QuestionEngine tests", () => {
+  test("1: Initial question and answer will initialize to empty answer array and highup and lowdown to undefined.", () => {
+    const q = TestDataFactory.createQuestionLaterTitrate();
+    const state = {
+      QandA: [QuestionAndAnswer.create(q)],
+      currentQuestionIdx: 0,
+    };
+    expect(state.QandA[0].answers).not.toBeUndefined();
+    expect(state.QandA[0].answers.length).toBe(0);
+    expect(state.QandA[0].highup).toBeUndefined();
+    expect(state.QandA[0].lowdown).toBeUndefined();
+  });
+
+  test("startSurvey should create a single answer entry for titration question.", () => {
+    const state = {
+      QandA: [
+        QuestionAndAnswer.create(
+          TestDataFactory.createQuestionLaterTitrate(),
+          1000,
+          undefined
+        ),
+      ],
+      currentQuestionIdx: 0,
+    };
+    const qe = new QuestionEngine();
+    qe.startSurvey(state);
+    expect(state.currentQuestionIdx).toBe(0);
+    expect(state.QandA[0].answers).not.toBeUndefined();
+    expect(state.QandA[0].answers.length).toBe(1);
+    expect(state.QandA[0].answers[0].amountEarlier).toBe(500);
+    expect(state.QandA[0].answers[0].timeEarlier).toBe(1);
+    expect(state.QandA[0].answers[0].amountLater).toBe(1000);
+    expect(state.QandA[0].answers[0].timeLater).toBe(3);
+    expect(state.QandA[0].highup).toBe(500);
+    expect(state.QandA[0].lowdown).toBeUndefined();
+  });
+
+  test("startSurvey should create a single answer entry for non titraiton question.", () => {
+    const state = {
+      QandA: [
+        QuestionAndAnswer.create(
+          TestDataFactory.createQuestionNoTitrate(),
+          undefined,
+          undefined
+        ),
+      ],
+      currentQuestionIdx: 0,
+    };
+    const qe = new QuestionEngine();
+    qe.startSurvey(state);
+    expect(state.currentQuestionIdx).toBe(0);
+    expect(state.QandA[0].answers).not.toBeUndefined();
+    expect(state.QandA[0].answers.length).toBe(1);
+    expect(state.QandA[0].answers[0].amountEarlier).toBe(400);
+  });
+
+  test("Test example for the values from Read 2001 paper.", () => {
```

Comment:
will keep looking at this today, but if I don't get through this maybe you could walk me through this test during our meeting today.

Reply:
When we meet next, I can walk you through the paper calculations.

Reply:
That test file is empty.  I put a dummy test in it for now.  I can delete it.  I was planning on adding some tests for loading the data; however, that is not a high priority now.